include/rcutils/error_handling.h

```diff
@@ -223,6 +223,20 @@ rcutils_set_error_state(const char * error_string, const char * file, size_t lin
     } \
   } while (0)
 
+/// Indicate that the function intends to set an error message and return an error value.
```

Comment:
I would add some comment mentioning that this is part of the fault injection so people understand its intended purpose.